atuin/src/command/client/history.rs

```diff
@@ -176,63 +176,127 @@ pub fn print_cmd_only(w: &mut StdoutLock, h: &[History]) {
 }
 
 impl Cmd {
-    pub async fn run(&self, settings: &Settings, db: &mut impl Database) -> Result<()> {
-        let context = current_context();
+    async fn handle_start(
```

Comment:
I've split `run` into private sub-fns, because it grew larger than clippy limit :)